ember-resources/src/util/function.ts

```diff
@@ -92,10 +93,8 @@ export function trackedFunction<Return>(...passedArgs: UseFunctionArgs<Return>)
     fn = passedArgs[2];
   }
 
-  return resource<State<Return>>(context, (hooks) => {
-    let state = new State(initialValue);
-
-    (async () => {
+  return resource<TrackedFunctionProperty<Return>>(context, (hooks) => {
+    const getValue = async (state: State<Return>) => {
```

Comment:
Can this function move inside the new class?

Reply:
Done